src/mumble/main.cpp

```diff
@@ -671,19 +675,29 @@ int main(int argc, char **argv) {
 }
 
 #if defined(Q_OS_WIN) && defined(QT_NO_DEBUG)
-extern void qWinMain(HINSTANCE, HINSTANCE, LPSTR, int, int &, QVector<char *> &);
-
 extern "C" __declspec(dllexport) int MumbleMain(HINSTANCE instance, HINSTANCE prevInstance, LPSTR cmdArg, int cmdShow) {
-	Q_UNUSED(cmdArg);
+	Q_UNUSED(instance)
+	Q_UNUSED(prevInstance)
+	Q_UNUSED(cmdArg)
+	Q_UNUSED(cmdShow)
+
+	int argc;
+	wchar_t **argvW = CommandLineToArgvW(GetCommandLineW(), &argc);
+	if (argvW == Q_NULLPTR) {
+		return -1;
+	}
+
+	QVector<QByteArray> argvS;
+	argvS.reserve(argc);
 
-	QByteArray cmdParam = QString::fromWCharArray(GetCommandLine()).toLocal8Bit();
-	int argc = 0;
+	QVector<char *> argvV(argc, Q_NULLPTR);
+	for (int i = 0; i < argc; ++i) {
+		argvS.append(QString::fromWCharArray(argvW[i]).toLocal8Bit());
```

Comment:
[toLocal8Bit](https://doc.qt.io/qt-5/qstring.html#toLocal8Bit) says this may be undefined after conversion? Any way around that?

Reply:
Unfortunately not, the ideal solution would be to use Unicode arguments directly:
https://bugreports.qt.io/browse/QTBUG-3200
https://bugreports.qt.io/browse/QTBUG-46118

Reply:
In that case you could use `toUtf8`.

I'm not sure main works fine with that Unicode though.